File: utils/dockerfiles/Dockerfile.cpp
# Use an official GCC compiler as a parent image
FROM gcc:latest

# Set the working directory
WORKDIR /app

# Use a script to compile and run the C++ code
COPY cpprunner.sh /usr/local/bin/cprun
RUN sed -i 's/\r$//' /usr/local/bin/cprun
RUN chmod +x /usr/local/bin/cprun

# Set the entry point to the run script
ENTRYPOINT ["cprun"]

File: utils/dockerfiles/Dockerfile.java
# Use an official OpenJDK image as a parent image
FROM openjdk:11

# Set the working directory inside the container
WORKDIR /app

# Copy the Java code to the working directory

COPY javarunner.sh /usr/local/bin/javarunner
RUN sed -i 's/\r$//' /usr/local/bin/javarunner
RUN chmod +x /usr/local/bin/javarunner

# Set the entry point to a custom script to compile and run the Java code
ENTRYPOINT ["bash", "javarunner"]
